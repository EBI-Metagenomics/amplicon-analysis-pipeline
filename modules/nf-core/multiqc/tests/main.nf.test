nextflow_process {

    name "Test Process MULTIQC"
    script "../main.nf"
    process "MULTIQC"

    tag "modules"
    tag "modules_nfcore"
    tag "multiqc"

    config "./nextflow.config"

    test("multiqc - report containing just fastp") {

        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file('${moduleDir}/tests/ERR4334351.fastp.json', checkIfExists: true)
                ]
                input[1] = []
                input[2] = []
                input[3] = []
                input[4] = []
                input[5] = []
                input[6] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("multiqc_versions_single") },
                { assert process.out.report[0] ==~ ".*/multiqc_report.html" },
                { assert process.out.data[0] ==~ ".*/multiqc_data" }
            )
        }

    }
}
