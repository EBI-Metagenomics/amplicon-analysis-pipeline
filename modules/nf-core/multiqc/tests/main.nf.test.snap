{
    "multiqc_versions_single": {
        "content": [
            [
                "versions.yml:md5,6eb13f3b11bbcbfc98ad3166420ff760"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-10T12:41:34.562023"
    }
}